pkgs/development/python-modules/pycparser/default.nix

```diff
@@ -22,6 +23,10 @@ buildPythonPackage rec {
   nativeCheckInputs = [ unittestCheckHook ];
   disabled = pythonOlder "3.8";
 
+  preCheck = lib.optionalString stdenv.cc.isClang ''
```

Comment:
Do the tests work on Darwin? If so then we need to exempt Darwin.

Reply:
Do we though?  Trying to call `gcc` when we known we're on a clang stdenv seems to be the wrong thing to do in all cases.

Reply:
The tests do appear to run on [`x86_64-darwin`](https://hydra.nixos.org/job/nixpkgs/trunk/python312Packages.pycparser.x86_64-darwin) and [`aarch64-darwin`](https://hydra.nixos.org/job/nixpkgs/trunk/python312Packages.pycparser.aarch64-darwin). We can prevent unnecessary rebuilds by adding `!stdenv.hostPlatform.isDarwin`.

Reply:
This is aimed at staging, it will be rebuilt anyway.

Reply:
It doesn't need to if the `prePatch` is applied in an `optionalAttrs` case. I'd like to avoid pushing to staging as much as possible.

Reply:
We should only be using conditionals (and therefore increasing variance between platforms) where it's necessary, which it isn't here.  Skipping staging by introducing a conditional as a convenience is fine, but only if somebody takes responsibility for the conditional being immediately removed on staging after the PR is merged, so we don't have that long term divergence.

Reply:
How exactly does it diverge? I don't see how it causes a divergence. Also, don't we have many conditionals aside from LLVM stuff which causes divergences?

Reply:
It diverges because the replacement is only done sometimes, when it would be find for it to always happen.  Always doing it means it won't break if somebody didn't test the with a platform that satisfied the condition, making it more likely it will keep working in the future.

> Also, don't we have many conditionals aside from LLVM stuff which causes divergences?

Yes, Nixpkgs is a codebase which has been around for 20 years and as a result has some code in it which does not follow best practices.  This is not a justification to add more.